Include/object.h

```diff
@@ -112,16 +112,28 @@ typedef struct _object {
     struct _typeobject *ob_type;
 } PyObject;
 
-/* Cast argument to PyObject* type. */
-#define _PyObject_CAST(op) ((PyObject*)(op))
+/* Cast argument to PyObject* type.
+
+   Raise a compilation error if the argument type is PyObject or PyObject**.
+
+   Detect for example the bug "&item" instead of "item" in:
+
+   const char *func(PyObject *item) { return Py_TYPE(&item)->tp_name; }
+   */
+#define _PyObject_CAST(op) \
+    ((PyObject*)(op) + Py_BUILD_ASSERT_EXPR(sizeof(*op) != sizeof(op)))
```

Comment:
if ``op`` is an expression, then your dereferencing is likely to fail, better use `*(op)`.

Reply:
And you don't have the issue of `op` being evaluated three times because sizeof does not evaluate the expression \o/

Reply:
Oh right, fixed.